### ISSUE TITLE
TSC does not crosscompile currently

### ISSUE THREAD

**Issue by Quintus** [Secretchronicles/TSC#531]
The build system for crosscompilation has not yet been adapted for the SFML changes, thus TSC currently doesn’t crosscompile to Windows.

Since with #10 there’s another large dependency task waiting to be done, this shouldn’t be targetted until #10 is completed.

Valete,
Quintus

**Comment by datahead8888** [comment]
Do we still intend to put the SFML changes in their own release before releasing the CEGUI conversion?  If so, we would probably want to get cross compilation working again earlier.

**Comment by Quintus** [comment]
No Windows crosscompiled build until this is resolved: https://github.com/mxe/mxe/pull/1447

Vale,
Quintus

**Comment by datahead8888** [comment]
Oh, this is very impacting to the next release...

**Comment by Quintus** [comment]
We can just compile it ourselves for the Windows build and link it dynamically (which is recommended by the CEGUI docs anyway). For the Linux build, we'll simply depend on the distro's library as usual. If the above ticket gets an answer by the MXE team, we can still remove our custom CEGUI build again.

Vale,
Quintus

**Comment by Quintus** [comment]
Compiling CEGUI is a joy. So far, I have created four bug reports on the CEGUI build system and one against MXE.

* https://github.com/mxe/mxe/issues/1463
* https://bitbucket.org/cegui/cegui/issues/1131/opengl-renderers-fail-to-crosscompile-due
* https://bitbucket.org/cegui/cegui/issues/1132/libxml2-based-xml-parser-fails-to
* https://bitbucket.org/cegui/cegui/issues/1133/tinyxml-based-xml-parser-fails-to
* https://bitbucket.org/cegui/cegui/issues/1134/ceguibase-library-fails-to-crosscompile

I know, I can't touch things without breaking them. I don't know why this happens!

Valete,
Quintus

**Comment by Quintus** [comment]
Okay. One of the CEGUI bugs has turned out to be a bug in CMake itself. The CEGUI devs are going to report that one upstream, I'll follow up here if I get a ticket URL.

Oh, and here's another bug I opened against CEGUI: https://bitbucket.org/cegui/cegui/issues/1135/pkg-config-files-not-installed-when-using

I also trigged yet another bug in MXE: https://github.com/mxe/mxe/pull/1465

I am really good at breaking things.

**Comment by Quintus** [comment]
The MXE team has merged in my PR for upgrading to CEGUI 0.8.x and the CEGUI team resolved the most pressing issues in CEGUI's source tree now. The MXE devs have MXE now track CEGUI's development branch `v0-8` instead of the stable release, so MXE is directly usable for compiling TSC.

I have updated [our MXE fork](https://github.com/Secretchronicles/mxe) and removed the old patches we had in that repository. Check out the `tsc-crosscompile` branch from that repository to get the exact MXE commit I have been building with.

As of commit  021de2827536d15adfca0773119de2a04d93f045, TSC now crosscompiles successfully to Windows and starts. When generating the image cache however, it causes a segmentaiton fault. This will have to be fixed, but has most likely nothing to do with the build system.

I have updated INSTALL.md accordingly.

Also note that due to [this MXE bug](https://github.com/mxe/mxe/issues/1351) it is currently not possible to build MXE on ArchLinux. All Linux distributions that ship a GCC < 6.0.0 should work fine.

Valete,
Quintus
